### DISCUSSION TITLE
About the latency of audio interfaces

### DISCUSSION THREAD

**Discussion by opposj** [Milkitic/KeyASIO.Net#4] [General]
I've observed an unnormal case that the Motu M2 seems to have more latency than the Scarlett 2i2 in KeyAsio, but its measured RTL is much lower than the Scarlett 2i2's:

The Motu M2
![Motu M2](https://user-images.githubusercontent.com/49869100/180920478-a45f249e-0df3-4a96-8854-2a2f49c9e35f.PNG)

The Scarlett 2i2
![2i2](https://user-images.githubusercontent.com/49869100/180920490-cfe5b372-0fb4-46de-892f-629c32852606.PNG)

I think 2.4ms RTL is reasonable for Motu M2 as it is roughly consistent with the official report, which is 2.5ms. In such case, the measured latency in KeyAsio should be about 1.2ms? But it is 1.8ms or so😢, quite frustrating anyway. In addition, for the Scarlett 2i2, its measured RTL, the RTL reported in the driver and the latency in KeyAsio are all consistent. So what's wrong with the Motu M2? Is it because I bought the wrong one, or the problem of the driver or firmware version?

--------------------------------------------------------------------------------------------------------------------------------------------------------

It seems that I have found out the reason,
![Motu M2 Details](https://user-images.githubusercontent.com/49869100/180928245-60a64811-6ac5-44d6-8041-472abbe6f305.PNG)

The output latency of Motu M2 is three times the latency of the input😅. So is it because of the driver or firmware version?

**Comment by Milkitic** [comment]
In KeyAsio the value is just simply multiplied by this formula: https://github.com/Milkitic/KeyASIO.Net/blob/master/KeyAsio.Gui/Converters/Multi_LatencySampleRate2LatencyConverter.cs#L13
So it is not a measured value, but a computed value from ASIO driver which is a sketchy latency.

**Reply by opposj** [reply]
So, the "playbackLatency" in the formula is simply fetched from the driver information? Whatever, it seems that the output latency of the Motu M2 is actually, hmmm..., higher than I expect. I think the Scarlett 2i2 may be a much better choice for Osu! due to its better output latency and lower price😋
By the way, in "Auto"/"No Fail" mode, I seem to only hear the key sound at the end of the slider most of the time, after adjusting the offset of KeyAsio to about -40, I can occasionally hear the key sound of all notes and sliders, is this normal?
